pkg/controller/components/clusterstatusupdate.go

```diff
@@ -72,8 +73,9 @@ func (c *clusterStatusUpdateComponent) Reconcile(ctx *core.Context) (ctrl.Result
 		csc.WorkerSelector = selector.String()
 		modified = true
 	}
-	if csc.WorkerReplicas != *sts.Spec.Replicas { // NOTE: panic: runtime error: invalid memory address or nil pointer dereference
-		csc.WorkerReplicas = *sts.Spec.Replicas
+	replicas := pointer.Int32Deref(sts.Spec.Replicas, 0)
+	if csc.WorkerReplicas != replicas { // NOTE: panic: runtime error: invalid memory address or nil pointer dereference (see DOM-40865)
+		csc.WorkerReplicas = replicas
```

Comment:
```suggestion
	if replicas := pointer.Int32Deref(sts.Spec.Replicas, 0); csc.WorkerReplicas != replicas {
		csc.WorkerReplicas = replicas
```

Reply:
Any reason not to use a scoped predicate variable here? Also, in my mind, the comment about panics is not terribly relevant at this piont because it's obvious that a nil pointer will panic when accessed, but this no longer has that issue.